### PR TITLE
add support for unmapped BAM file reading

### PR DESCRIPTION
**Pull request by danielmarrama**
For issue #32 

`sam.reader` takes unmapped bool and passes to `_pysam_open`

```python
    return _pysam_open(
        path=path, open_for_reading=True, file_type=file_type, unmapped=unmapped
    )
```

Then `kwags["check_sq"] = False` added if unmapped and passed to `pysam.AlignmentFile`

### CODE REVIEW

File: fgpyo/sam/__init__.py [hunk 1]
```diff
@@ -219,8 +219,8 @@ def from_path(cls, path: Union[Path, str]) -> "SamFileType":
 
 
 def _pysam_open(
-    path: SamPath, open_for_reading: bool, file_type: Optional[SamFileType] = None, **kwargs: Any
-) -> SamFile:
+    path: SamPath, open_for_reading: bool, file_type: Optional[SamFileType] = None, 
+    unmapped: bool = False, **kwargs: Any) -> SamFile:
```
Comment (nh13) [inline fgpyo/sam/__init__.py:223]: don't forget to add `unmapped` to the docstring.

**Review by nh13 — CHANGES_REQUESTED** [review]
Could you add a quick unit test to test this?

File: fgpyo/sam/__init__.py (hunk 1 above)
Comment (danielmarrama) [reply fgpyo/sam/__init__.py:223]: Yes, I forgot that. I'll add a test as well.

**Comment by danielmarrama** [comment]
Added `unmapped.sam` file and unit test `test_unmapped_sam_file_open_reading_with_reader` in [test/test_sam.py](https://github.com/fulcrumgenomics/fgpyo/blob/main/fgpyo/sam/tests/test_sam.py).  Behavior should be the same with both SAM and BAM files, right? I can add another test checking BAM file reading if necessary.

**Review by nh13 — APPROVED** [review]

**Comment by nh13** [comment]
Thanks @danielmarrama !!!

**Pull request merged**